src/staking/stakes_db.cpp

```diff
@@ -500,14 +499,24 @@ bool CStakesDBCache::registerFreeTransaction(const CScript& script, const uint32
         freeTxInfo = getFreeTxInfoForScript(script);
     }
     assert(freeTxInfo.isValid());
+    if (nHeight > 0 && freeTxInfo.getCurrentWindowStartHeight() == 0) {
+        freeTxInfo.setCurrentWindowStartHeight(nHeight);
+    }
     std::set<uint256> activeStakeIdsChain = getActiveStakeIdsForScript(script);
     if (activeStakeIdsChain != freeTxInfo.getActiveStakeIds()) {
         freeTxInfo.setActiveStakeIds(activeStakeIdsChain);  //TODO(mtwaro): recalculate limit
     }
-    if (freeTxInfo.getUsedLimit() + txSize > freeTxInfo.getLimit()) {
-        return false;
+    if (nHeight == 0) {
+        if (!freeTxInfo.addUnconfirmedTxId(tx.GetHash(), tx.GetTotalSize())) {
```

Comment:
can be put into one if nHeight == 0 && (!freeTxInfo.addUnconfirmedTxId(tx.GetHash(), tx.GetTotalSize()))